test/unit_tests/bloc_tests/search_bloc_test.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mobile_app/src/blocs/search_bloc/search_bloc.dart';
+import 'package:mobile_app/src/blocs/search_bloc/search_event.dart';
+import 'package:mobile_app/src/blocs/search_bloc/search_state.dart';
+import 'package:mobile_app/src/constants/constant_text.dart';
+import 'package:mobile_app/src/models/recipe_model.dart';
+import 'package:mobile_app/src/services/search_services.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../cloud_firestore_mock.dart';
+
+final String fakeValidSearchValue = 'Chicken';
+final String fakeNoResultValue = 'Fake no result';
+final fakeRecipes = [
+  RecipeModel(name: 'Chicken 1'),
+  RecipeModel(name: 'Chicken 2'),
+];
+
+class MockSearchServices extends Mock implements SearchServices {
+  @override
+  Future<List<RecipeModel>> searcRecipesByName(String searchQuery) async {
+    if (searchQuery == fakeValidSearchValue) {
+      return fakeRecipes;
+    }
+    if (searchQuery == fakeNoResultValue) {
+      return [];
+    }
+    throw Exception();
+  }
+}
+
+void main() {
+  late MockSearchServices mockSearchServices;
+  late SearchBloc searchBloc;
+
+  setUpAll(() async {
+    setupCloudFirestoreMocks();
+    await Firebase.initializeApp();
+  });
+
+  setUp(() {
+    mockSearchServices = MockSearchServices();
+    searchBloc = SearchBloc(searchServices: mockSearchServices);
+  });
+
+  tearDown(() {
+    searchBloc.close();
+  });
+
+  blocTest('emits [] when no event is called',
+      build: () => searchBloc, expect: () => []);
+
+  blocTest(
+      'emits [SearchTextFieldChangeSuccess] when [SearchTextFieldChanged] is called and text field is empty',
```

Comment:
Why logic emits success but text field empty ? When success it's mean already data right ?

Reply:
This state is **SearchTextFieldChangeSuccess** which means whenever user stop typing, it receive a new value of the text filed. Just imagine like you type in 'chicken', it update **recipeTextFieldValue** to be 'chicken' and if you delete them all later, the new value of course will be empty. Maybe you misunderstood this state with **SearchRecipeSuccess(recipes: [])**.

Reply:
okay, I got that, you can skip it.

Reply:
so do we need to call api with empty query?

Reply:
No, I never did that. If the text field is empty, then **no** search request will be sent at all.